[14.0][MIG] l10n_es_ticketbai_api

ao-landoo opened the pull request:
Migración estándar a V14

ljsalvatierra-binovo committed:
[ADD] l10n_es_ticketbai_api: send invoices to the Basque Country tax agencies (API).
Base addon for l10n_es_ticketbai.
Added Python library dependencies to requirements.txt.

ao-landoo committed:
[MIG] l10n_es_ticketbai_api: Migration to 12.0

oca-travis committed:
[UPD] Update l10n_es_ticketbai_api.pot

ao-landoo committed:
[FIX] fix l10n_es_ticketbai_api module warnings

oca-travis committed:
[UPD] Update l10n_es_ticketbai_api.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-12.0/l10n-spain-12.0-l10n_es_ticketbai_api
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-12-0/l10n-spain-12-0-l10n_es_ticketbai_api/

ao-landoo committed:
[MIG] l10n_es_ticketbai_api: Migration to 13.0

ao-landoo committed:
[FIX] l10n_es_ticketbai_api: state field label, l10n_es_ticketbai: installation wizard info

oca-travis committed:
[UPD] Update l10n_es_ticketbai_api.pot

OCA-git-bot committed:
l10n_es_ticketbai_api 12.0.0.1.1

ao-landoo committed:
[IMP] l10n_es_ticketbai_api: black, isort, prettier

oca-travis committed:
[UPD] Update l10n_es_ticketbai_api.pot

ao-landoo committed:
[MIG] l10n_es_ticketbai_api: Migration to 14.0

gelo-landoo reviewed (approved):
Aprobación funcional

ao-landoo committed:
[FIX] l10n_es_ticketbai_api: fix execution warnings

oca-travis committed:
[UPD] Update l10n_es_ticketbai_api.pot

OCA-git-bot committed:
l10n_es_ticketbai_api 13.0.1.0.1

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-13.0/l10n-spain-13.0-l10n_es_ticketbai_api
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-13-0/l10n-spain-13-0-l10n_es_ticketbai_api/

Bilbonet reviewed (approved):
Instalación y configuración correctas.
Anoto aquí los warnings  por si los quieres revisar para alguna modificacion:

WARNING  odoo.addons.base.models.ir_model: Two fields (test_qr_base_url, qr_base_url) of tbai.tax.agency() have the same label: QR Base URL.

WARNING  odoo.models: method res.company._check_tbai_developer_id: @constrains parameter 'tbai_developer_id' is not writeable

WARNING odoo.models: method res.company._check_tbai_license_key: @constrains parameter 'tbai_license_key' is not writeable

WARNING odoo.models: method res.company._check_tbai_software_name: @constrains parameter 'tbai_software_name' is not writeable

ao-landoo commented:
> Instalación y configuración correctas.
> Anoto aquí los warnings por si los quieres revisar para alguna modificacion:
> 
> WARNING odoo.addons.base.models.ir_model: Two fields (test_qr_base_url, qr_base_url) of tbai.tax.agency() have the same label: QR Base URL.
> 
> WARNING odoo.models: method res.company._check_tbai_developer_id: @constrains parameter 'tbai_developer_id' is not writeable
> 
> WARNING odoo.models: method res.company._check_tbai_license_key: @constrains parameter 'tbai_license_key' is not writeable
> 
> WARNING odoo.models: method res.company._check_tbai_software_name: @constrains parameter 'tbai_software_name' is not writeable

Gracias por el aviso Jesus! Tengo mas de un pequeño cambio pendiente antes de mergearlo.

ao-landoo committed:
[13.0][FIX] l10n_es_ticketbai_api: create partners from test file

OCA-git-bot committed:
l10n_es_ticketbai_api 13.0.1.0.2

xAdrianC-Kernet reviewed (approved):
Instalación y configuración ok

ao-landoo committed:
[IMP] l10n_es_ticketbai_api: black, isort, prettier

ao-landoo commented:
@pedrobaeza Se puede mergear? Gracias

pedrobaeza commented:
Aparece el siguiente warning en runbot:
```
2021-07-07 06:57:42,153 167 WARNING openerp_test py.warnings: /.repo_requirements/virtualenv/python3.6/lib/python3.6/site-packages/requests_pkcs12.py:20: DeprecationWarning: PKCS#12 support in pyOpenSSL is deprecated. You should use the APIs in cryptography.
from OpenSSL.crypto import load_pkcs12, dump_certificate, dump_privatekey, FILETYPE_PEM
```

¿No estáis usando la gestión de certificados de `l10n_es_aeat` y habéis duplicado el código? En cualquier caso, en #1705 tenéis cómo convertirlo.

ao-landoo commented:
@pedrobaeza He eliminado la dependencia a la librería pyOpenSSL pero veo en Travis que sigue apareciendo el warning, entiendo que es porque también tengo que eliminar la librería `requests_pkcs12`.

¿Es correcto?

ao-landoo commented:
@pedrobaeza La librería `requests_pkcs12` utiliza también pyOpenSSL, por eso sigue apareciendo el warning. He ido al repositorio de la librería y veo que tienen un PR abierto para quitar la dependencia de pyOpenSSL.

¿Podemos mergear como está ahora el PR? Cuando `requests_pkcs12` deje de depender de pyOpenSSL actualizaré esté módulo para que exija la última versión de la librería.

[PR `requests_pkcs12`](https://github.com/m-click/requests_pkcs12/pull/28)

pedrobaeza commented:
Por qué no cambiais a la gestión de certificados de `l10n_es_aeat`, y si no, al menos utilizad el mismo tipo de código para evitar ese warning de obsolescencia.

ao-landoo commented:
@pedrobaeza Se decidió que TicketBAI API no tenga dependencias a `l10n_es_aeat`, entonces por ahora lo mantendremos así. He eliminado la dependencia de `requests_pkcs12` y ya no hay warnings. ¿Se puede mergear?

pedrobaeza commented:
/ocabot merge nobump
